### PR TITLE
Add course link and "..." dropdown to nav bar

### LINKED ISSUE #2955: Educational Course Connection
With the new educational course coming out at the end of the quarter, we need to determine how people will access it from our websites.  

Each website will access the same school: Iterative Tools School.  

I would like for the school to be accessible  from the navigation and in DVC.org's case, from the Community Page.  

[Here is information on how their domains work](https://support.learnworlds.com/support/solutions/articles/12000080095-general-overview-custom-domains#What-is-a-custom-domain?). Let me know how you think it's best to proceed!

@julieg18 @rogermparent @yalozhkin

### PR DESCRIPTION
**Pull request by julieg18**
* refactors nav popup code
* adds `...` dropdown to nav bar (holds items that don't fit on the nav bar)
* add "Course" nav item

Fixes #2955

### CODE REVIEW

File: src/components/LayoutHeader/Nav/LinkItems/index.tsx [hunk 1]
```diff
@@ -45,158 +55,91 @@ const navLinkItemsData: Array<INavLinkData | INavLinkPopupData> = [
...
-    if (isOtherToolsPopupOpen) {
-      closeAllPopups()
-    } else {
-      openOtherToolsPopup()
-    }
+): item is INavLinkPopupData =>
+  (item as INavLinkPopupData).popupName !== undefined
+
+const LinkItems: React.FC = () => {
+  const communityPopup = usePopup()
```
Comment (julieg18) [inline src/components/LayoutHeader/Nav/LinkItems/index.tsx:91]: I went ahead and created a `usePopup` hook that handles popup functionality. The popup code in the nav was already overly complex and adding a third popup would have made things even more complicated 😅

File: src/components/LayoutHeader/Nav/LinkItems/styles.module.css [hunk 2]
```diff
@@ -66,6 +78,18 @@
     }
   }
 
+  &.other {
```
Comment (julieg18) [inline src/components/LayoutHeader/Nav/LinkItems/styles.module.css:81]: Here's what the popup looks like: 

image


Is it good enough for a first iteration? cc @yalozhkin

File: src/components/LayoutHeader/Nav/LinkItems/index.tsx [hunk 3]
```diff
@@ -1,33 +1,43 @@
...
   text: string
-  popup: PopupName
+  popupName: PopupName
+  Popup: React.FC<IPopupProps>
+  className?: string
+  href?: string
 }
 
-const navLinkItemsData: Array<INavLinkData | INavLinkPopupData> = [
+export const navLinkItemsData: Array<INavLinkData | INavLinkPopupData> = [
```
Comment (julieg18) [inline src/components/LayoutHeader/Nav/LinkItems/index.tsx:40]: I wasn't sure where "Learn" should go in the nav. For now, I placed it after "Support" :)

**Review by rogermparent — APPROVED** [review]
Looks fine by me, especially as a first iteration before we try unifying all the navs.

Are we okay with "Community" losing its dropdown when in this menu? I don't see a problem with it, but others may think differently. That said, if we want to change it we can do so in a follow up, since this is still a step up from before where it wasn't present at all!

**Comment by shcheklein** [comment]
Thanks, @julieg18 ! Can we move the Learn link to the Docs., Blog? (unless you discussed this other folks and this decision was made based of that). It's not a strong opinion, but it makes sense to me to keep them nearby.

**Comment by julieg18** [comment]
> Thanks, @julieg18 ! Can we move the Learn link to the Docs., Blog? (unless you discussed this other folks and this decision was made based of that). It's not a strong opinion, but it makes sense to me to keep them nearby.

If I'm understanding your question correctly, you want to move it to after Blog? I moved it there but let me know if I misunderstood :)

> We already have Learn under the community dropdown, so we should consider changing either of these.

You're right! Plus, as we mentioned before, Learn is kind of an unclear name anyway. I renamed it to "Course" for now! 

> I think we also need to find a proper place on the hamburger menu as well.

Thanks for catching that! I'll update it.

> Q: should we consider adding a new window SVG icon similar to what we have with items on other tools menus?

I'm not sure I understand your question 🤔 Where exactly do you want to add a new window SVG icon?

**Comment by yathomasi** [comment]
> I'm not sure I understand your question 🤔 Where exactly do you want to add a new window SVG icon?

screenshot-dvc-org-add-course-conn-nsrmv2 herokuapp com-2022 02 09-14_03_38

I meant something like this like we are using on `Other tools`. But, I am not sure we would want that.

**Comment by julieg18** [comment]
> I meant something like this like we are using on Other tools. But, I am not sure we would want that.

Oh, ok! Thanks for clarifying further! Personally, I don't think it's needed. But @iterative/websites, what do you think?

**Comment by shcheklein** [comment]
Yeah, good point @yathomasi , but I also don't think we need it in this specific case. I would consider adding the link only if it's a completely external website, in this case it's kinda internal link (ecosystem of the websites), so it's fine to avoid it here. And probably in general, menu is to tight on space that it makes sense to sacrifice some consistency here.

**Pull request merged**